src/dlsnode/storage/BucketFile.d

```diff
@@ -334,6 +343,70 @@ public class BucketFile: OutputStream
         }
     }
 
+
+    /**************************************************************************
+
+        Opens the bucket file and reads or writes the bucket header in a
+        non-blocking fashion.
+
+        Params:
+            path = file path
+            suspended_job = JobNotification to block
+                the fiber on until read is completed
+            file_buffer = buffer used for buffered input
+            style = style in which file will be open for
+
+    **************************************************************************/
+
+    public bool openAsync (cstring path,
+            JobNotification suspended_job,
+            void[] file_buffer,
+            File.Style style = File.ReadExisting)
+    in
+    {
+        assert(style == File.ReadExisting || style == File.ReadWriteAppending,
+                "Currently BucketFile only supports ReadExisting or ReadWriteAppending");
+    }
+    body
+    {
+        if (!this.is_open_ || style == File.ReadWriteAppending)
```

Comment:
Is there a use case for opening an already open file with `ReadWriteAppending`?

Reply:
Good point.